### PR TITLE
Problem: omni_httpd content-length behavior is not clear

### PR DESCRIPTION
**Pull request by steve-chavez**
omni_httpd currently sets it automatically for any non-null response body.

Solution: document it and add a test for null response body behavior.

----

Related to the discussion on https://github.com/omnigres/omnigres/issues/279#issuecomment-1784135868

### CODE REVIEW

**Comment by yrashk** [comment]
Thank you so much for your contribution! 🤟 

Resuming our conversation in #279:

1. Do we want Content-Length to be absent if the body is `null`? 
2. As for the overflow, I am trying to think if this is more of a warning or error. I am inclined to believe this is a warning as we don't do anything crazy there, but it is indicative of a potential problem one wouldn't want to bury. I believe `omni_alerts` (when it comes to live) is the best way to alert ops of the problem, but whether that's a 500 or not, it is hard to say. Thoughts?

File: extensions/omni_httpd/docs/headers.md [hunk 1]
```diff
@@ -8,4 +8,73 @@ select omni_http.http_header_get(request.headers, 'host') as host;
...
+## Content-Length
+
+`omni_httpd` automatically sets the `Content-Length` header for any non-null response body.
+
+For instance, having:
+
+```sql
+create extension omni_httpd cascade;
+
+set omni_httpd.init_port to 8000;
```
Comment (yrashk) [inline extensions/omni_httpd/docs/headers.md:22]: This is ineffectual because only the extension installation script uses these parameters.

This can lead to confusion. Perhaps for documenting purpose, we can remove this (instead of moving it above `create extension`)

File: extensions/omni_httpd/docs/headers.md [hunk 2]
```diff
@@ -8,4 +8,73 @@ select omni_http.http_header_get(request.headers, 'host') as host;
 select omni_http.http_header_get_all(request.headers, 'accept') as accept;
 ```
 
-The header name these functions take is case-insensitive.
\ No newline at end of file
+The header name these functions take is case-insensitive.
+
+## Content-Length
```
Comment (yrashk) [inline extensions/omni_httpd/docs/headers.md:13]: Great idea to document this!

File: extensions/omni_httpd/docs/headers.md (hunk 1 above)
Comment (steve-chavez) [reply extensions/omni_httpd/docs/headers.md]: Removed. Now using the default `8080` port for the curl examples instead.

**Comment by steve-chavez** [comment]
> Do we want Content-Length to be absent if the body is null?

It seems a bit wasteful to always include a `Content-Length: 0` when the body is known to be null. Also, the [RFC](https://www.rfc-editor.org/rfc/rfc9110#name-content-length) doesn't mandate setting `Content-Length` for empty bodies. So any behavior would be correct, but the `null` to no `Content-Length` might be more intuitive.

> As for the overflow, I am trying to think if this is more of a warning or error. I am inclined to believe this is a warning as we don't do anything crazy there, but it is indicative of a potential problem one wouldn't want to bury. I believe omni_alerts (when it comes to life) is the best way to alert ops of the problem, but whether that's a 500 or not, it is hard to say. Thoughts?

Taking a step back, why is `Content-Length` overridable? What use case would that cover?

`omni_httpd` could always compute the correct `Content-Length` and if a user sets it, we could emit a WARNING saying "ignoring user defined content-length". To resize the response, users could manipulate the body directly.

**Comment by yrashk** [comment]
> > Do we want Content-Length to be absent if the body is null?
> 
> It seems a bit wasteful to always include a `Content-Length: 0` when the body is known to be null. Also, the [RFC](https://www.rfc-editor.org/rfc/rfc9110#name-content-length) doesn't mandate setting `Content-Length` for empty bodies. So any behavior would be correct, but the `null` to no `Content-Length` might be more intuitive.

Fair. I think we can agree on this.

> 
> > As for the overflow, I am trying to think if this is more of a warning or error. I am inclined to believe this is a warning as we don't do anything crazy there, but it is indicative of a potential problem one wouldn't want to bury. I believe omni_alerts (when it comes to life) is the best way to alert ops of the problem, but whether that's a 500 or not, it is hard to say. Thoughts?
> 
> Taking a step back, why is `Content-Length` overridable? What use case would that cover?
> 
> `omni_httpd` could always compute the correct `Content-Length` and if a user sets it, we could emit a WARNING saying "ignoring user defined content-length". To resize the response, users could manipulate the body directly.

This behavior was introduced very recently. Initially, I was handling a case when we had a case of HTTP handler producing its own content-length, and omni_httpd always set its own: 783ae3ade2bd99e96bf82f7b32421350dc29a9e4 and as a quick follow up, I wrote a test for it: e60deed42781c2e76ff393819fd72f25a7becfc3

The origin of this issue lies in frameworks such as, say, Flask, setting their own `content-length`. The reasoning I had was to basically respect what these frameworks say is the expected length and just as a precaution, resize the output when there's too much of it. May be that should also include a warning.

**Comment by steve-chavez** [comment]
> The origin of this issue lies in frameworks such as, say, Flask, setting their own content-length. The reasoning I had was to basically respect what these frameworks say is the expected length and just as a precaution, resize the output when there's too much of it. May be that should also include a warning.

I see, makes sense. It looks like this behavior is not something that the user might use directly through SQL. So I've removed the downsizing example and left overriding documented under a warning. Let me know what you think.

**Comment by yrashk** [comment]
I think it's perfect, at least for now! Thank you.
